<launch>

  <arg name="uav_odometry" value="uav_odometry"/>
  <arg name="reference_position" value="reference_position"/>

  <arg name="fa_disturbance_estimate" value="fa_disturbance_estimate"/>

  <arg name="load_odometry" value="load_odometry"/>

  <arg name="playing_back" default="false"/>
  <arg name="record" default="true" />

  <group unless="$(arg playing_back)">

    <node pkg="rosbag" type="record" name="rosbag_record"
      args="-o $(find quad_control)/experimental_data/rosbag/file /ServiceSequencer $(arg uav_odometry) uav_roll_pitch_yawrate_thrust $(arg fa_disturbance_estimate) $(arg reference_position)"
      if="$(arg record)"/>


    <!-- <include file="$(find quad_control)/launch/mav_with_load_example.launch"> -->
      <!-- all vars that included.launch requires must be set -->
      <!-- <arg name="uav_odometry" value="$(arg uav_odometry)"/> -->
      <!-- <arg name="load_odometry" value="$(arg load_odometry)"/> -->
    <!-- </include> -->

    <include file="$(find quad_control)/launch/firefly_example.launch">
      <arg name="uav_odometry" value="$(arg uav_odometry)"/>
    </include>

  </group>

  <param name="uav_odometry" type="str" value="$(arg uav_odometry)"/>
  <param name="reference_position" type="str" value="$(arg reference_position)"/>
  <param name="fa_disturbance_estimate" type="str" value="$(arg fa_disturbance_estimate)"/>
  <param name="load_odometry" type="str" value="$(arg load_odometry)"/>
  <param name="playing_back" type="bool" value="$(arg playing_back)"/>

    <!-- mission_type: may be specified by used when launching; or when this file is included elsewhere -->
    
    <arg name="gazebo" default="False"/>
    <arg name="mocap" default="False"/>
    <arg name="rviz" default="False"/>

    <node pkg="quad_control" name="service_time_sequencer" type="service_time_sequencer.py"  output="screen"/>


    <group if="$(arg gazebo)">
        <!-- define mission_type parameter: used to define mission database -->
        <param name="mission_type" type="str" value="gazebo"/>

        <include file="$(find quad_control)/launch/missions_gazebo.launch">
          <!-- all vars that included.launch requires must be set -->
          <!-- <arg name="hoge" value="fuga" /> -->
        </include>

    </group>

    <group if="$(arg mocap)">
        <!-- define mission_type parameter: used to define mission database -->
        <param name="mission_type" type="str" value="mocap"/>

        <include file="$(find quad_control)/launch/missions_mocap.launch">
          <!-- all vars that included.launch requires must be set -->
          <!-- <arg name="hoge" value="fuga" /> -->
        </include>

    </group>

    <group if="$(arg rviz)">
        <!-- define mission_type parameter: used to define mission database -->
        <param name="mission_type" type="str" value="rviz"/>

        <include file="$(find quad_control)/launch/missions_rviz.launch">
          <!-- all vars that included.launch requires must be set -->
          <!-- <arg name="hoge" value="fuga" /> -->
        </include>

        <!-- GUI -->
        <!-- <node name="rqt_gui" pkg="gui" type="tabbedGUI"/> -->
        <!-- <node name="rqt_gui" pkg="gui" type="tabPlugin"/> -->

        <!-- This runs rqt and we have to select plugin manually -->
        <!-- similar to rosrun rqt_gui rqt_gui -->
        <!-- <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui"/> -->
        <node name="rqt_gui" pkg="rqt_gui" type="rqt_gui" args="--standalone  tabbedGUI --args '' 'rviz' " output="screen"/>

        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find quad_control)/launch/rviz_config.rviz"/>

        <group ns="simulator">

          <node pkg="quad_control" name="service_time_sequencer" type="service_time_sequencer.py"  output="screen"/>

          <!-- Simulator Node -->
          <node pkg="quad_control" name="Simulator_Iris1" type="quad_simulator.py" output="screen"/>

          <!-- Simulator parameters -->
          <param name="gravity_sim" value="9.81"/>
          <param name="mass_quad_sim" value="1.442"/>
          <param name="Throttle_neutral_sim" value="1484.0"/>  
          <param name="ACRO_RP_P_sim" value="4.5"/> 
          <param name="ktt_inner_loop" value="10.0"/> 


          <!-- PARAMETERS THAT ARE COMMOM BETWEEN CONTROLLER AND SIMULATOR -->
          <!-- see WEB: ARDUCOPTER DIFFERENT MODES PARAMETERS -->
          <param name="MAX_ANGLE_DEG" value="45.0"/> 
          <param name="MAX_PSI_SPEED_Deg" value="200.0"/> 
        
        </group>  

    </group>        

</launch>